src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Utils.cs

```diff
@@ -742,8 +742,7 @@ internal static bool IsSelfSigned(X509Chain chain)
 
         internal static AsymmetricAlgorithm GetAnyPublicKey(X509Certificate2 certificate)
         {
-            // TODO: Add ?? certificate.GetDSAPublicKey(), when available (dotnet/corefx#11802).
-            return certificate.GetRSAPublicKey();
+            return (AsymmetricAlgorithm)certificate.GetRSAPublicKey() ?? certificate.GetDSAPublicKey();
```

Comment:
Yeah - when using the null-coalescing operator with 2 types that have a common base class, you have to explicitly cast one of them to disambiguate (otherwise it'll try to cast the second value to `RSA`, which is impossible).